include/libtorrent/torrent.hpp

```diff
@@ -478,26 +480,21 @@ namespace libtorrent
 
 		void stop_when_ready(bool b);
 
-		int started() const { return m_started; }
+		time_point started() const { return m_started; }
 		void step_session_time(int seconds);
 		void do_pause(bool clear_disk_cache = true);
 		void do_resume();
 
-		int finished_time() const;
-		int active_time() const;
-		int seeding_time() const;
+		seconds_t finished_time() const;
+		seconds_t active_time() const;
+		seconds_t seeding_time() const;
 
 		bool is_paused() const;
 		bool is_torrent_paused() const { return m_paused; }
 		void force_recheck();
 		void save_resume_data(int flags);
 
-		bool need_save_resume_data() const
-		{
-			// save resume data every 15 minutes regardless, just to
-			// keep stats up to date
-			return m_need_save_resume_data || m_ses.session_time() - m_last_saved_resume > 15 * 60;
-		}
+		bool need_save_resume_data() const;
```

Comment:
yes, we need to cast it because the diff duration type does not match to seconds32